### PR TITLE
Implement #7980: Dialog Framework: provide PhaseListener which allows using (JSF-)Flash with Dialog Framework

### LINKED ISSUE #7980: Dialog Framework: provide PhaseListener which allows using (JSF-)Flash with Dialog Framework
Flash can be used to pass data from one page to another.
On the way from a page to a dialog framework - page this is a bit more complicated. You need a PhaseListener which keep´s the Flash alive during opening the dialog framework - dialog. (eg DialogKeepFlashPhaseListener)
We already use such a PhaseListener inhouse and i´ll contribute it to PrimeFaces.

Known limitation:
Currently only works on Mojarra when combined with https://showcase.omnifaces.org/filters/GzipResponseFilter (there´s a bug within Mojarra which prevents it form working without OmniFaces GzipResponseFilter; i provided a PR https://github.com/eclipse-ee4j/mojarra/pull/4978 to fix this)

It work´s fine with MyFaces.

### PR DESCRIPTION
**Pull request by christophs78**

### CODE REVIEW

File: primefaces/src/main/java/org/primefaces/application/DialogKeepFlashPhaseListener.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    @Override
+    public PhaseId getPhaseId() {
+        return PhaseId.RENDER_RESPONSE;
+    }
+
+    private boolean isInDialogPreparation() {
+        String dialogOutcome = (String) FacesContext.getCurrentInstance().getAttributes().get(Constants.DialogFramework.OUTCOME);
+        return dialogOutcome != null && dialogOutcome.length() > 0;
```
Comment (tandraschko) [inline primefaces/src/main/java/org/primefaces/application/DialogKeepFlashPhaseListener.java]: use our LangUtils

File: primefaces-showcase/src/main/java/org/primefaces/showcase/view/df/DFLevel1View.java [hunk 2]
```diff
@@ -27,14 +27,30 @@
...
 @RequestScoped
 public class DFLevel1View {
 
+    private boolean flashChecked = false;
+
+    public void init() {
+        if (!flashChecked) {
+            LocalDateTime valueFromFlash = (LocalDateTime) FacesContext.getCurrentInstance().getExternalContext().getFlash().get("param1");
+            if (valueFromFlash != null) {
+                FacesContext.getCurrentInstance().addMessage("", new FacesMessage("value from flash",
```
Comment (melloware) [inline primefaces-showcase/src/main/java/org/primefaces/showcase/view/df/DFLevel1View.java]: Use `Constants.EMPTY_STRING` here

**Review by melloware — APPROVED** [review]

File: primefaces-showcase/src/main/java/org/primefaces/showcase/view/df/DFLevel1View.java (hunk 2 above)
Comment (tandraschko) [reply primefaces-showcase/src/main/java/org/primefaces/showcase/view/df/DFLevel1View.java]: please dont use PF internal constants in showcase

Comment (melloware) [reply primefaces-showcase/src/main/java/org/primefaces/showcase/view/df/DFLevel1View.java]: Yep just realized that was a showcase class.

**Comment by tandraschko** [comment]
can you also add a documentation in the DF part of the docs?

**Comment by christophs78** [comment]
> can you also add a documentation in the DF part of the docs?

yes, comes this evening

**Pull request merged**